pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -782,6 +785,16 @@ func getServersConfig(cr *api.KieApp) ([]api.ServerTemplate, error) {
 				template.Jvm = *serverSet.Jvm.DeepCopy()
 			}
 
+			if template.StartupStrategy == "" {
```

Comment:
I think the changes required here will be more complex than just it here, I would suggest you to follow the docs on to disable it and turn back to controller strategy and see the issues you'll face.